Change doc structure (#1825)
jupyterhub/zero-to-jupyterhub-k8s: doc/source/index.rst

`JupyterHub`_ allows users to interact with a computing environment through a
webpage. As most devices have access to a web browser, JupyterHub makes it is
easy to provide and standardize the computing environment for a group of people
(e.g., for a class of students or an analytics team).

This project will help you set up your own JupyterHub on a cloud/on-prem k8s environment and leverage
its scalable nature to support a large group of users. Thanks to
:term:`Kubernetes`, we are not tied to a specific cloud provider.

.. note::

   This project is under active development so information and tools may

>    change. *You can be a part of this change!* If you see anything that is incorrect
>    or have any questions, feel free to reach out in the `gitter chat
>    <https://gitter.im/jupyterhub/jupyterhub>`_ or create an issue at the `issues
>    page`_. If you have tips or deployments that you would like to share, see
>    :ref:`resources-and-references`.

Reviewer: We should mention Discourse too. 
(I'm not sure of the indentation of the suggestion here is ok, though :thinking: )
```suggestion
   change. *You can be a part of this change!* If you see anything that is incorrect, please create an issue at the
   `issues page`_. If you have any questions, feel free to post on the community `Discourse forum
   <https://discourse.jupyter.org/>`_ or reach out in the `Gitter chat <https://gitter.im/jupyterhub/jupyterhub>`_ or . If you
   have tips or deployments that you would like to share, see :ref:`resources-and-references`.
```

Author: Yes! Great addition, I did not notice that Discourse was not mentioned here.

   For a more elaborate introduction to the tools and services that JupyterHub
   depends upon, see our :ref:`tools` page.

.. _getting-to-zero:

> Setup Infrastructure

Reviewer: For me, this title was a bit vague and I believe adding "cluster" makes it a little bit clear what to expect.
(@consideRatio, what do you think?)
```suggestion
Setup Cluster Infrastructure
```

Reviewer 2: I find "Cluster" to be a overloaded word with too much different interpretations to be helpful, and that someone reading this without already assuming "Kubernetes cluster" when "cluster" is named would benefit from simply hearing "Setup Kubernetes" instead.

Setup Kubernetes / Setup Infrastructure - okay okay with both these options, but lean towards the first to be clearer.

Reviewer 2: Hmmm, I've been thinking about this a bit more, and think we should opt for Setup Kubernetes rather than Setup infrastructure. I'm worried that Setup Infrastructure implies that it will be in scope for this guide to help with for example setting up a dedicated out-of-Helm-chart database to reference in various cloud providers or similarly, but what we actually aim for have have chosen to limit the scope to, is to help setup the required k8s cluster to a functional state, which may of course involve some cloud specific details though.

Author: Sure! The only reason I had mentioned Setup Infra was to account for the Helm installation part mentioned. But, this will work out too, with a bit of context on the `Setup JupyterHub` landing page.

> ---------------------

Reviewer 2: ```suggestion
--------------------
```

This section describes a Kubernetes cluster and outlines how to complete *Step Zero: your Kubernetes cluster* for
different cloud providers and infrastructure.
Since this project utilizes Helm as the package manager for Kubernetes deployments defined as charts,
we will also cover how to set it up for your infrastructure.

.. toctree::
   :maxdepth: 2

   infra/index

.. _creating-your-jupyterhub:

Setup JupyterHub
----------------

This tutorial starts from *Step Zero: Your Kubernetes cluster* and describes the
steps needed for you to create a complete initial JupyterHub deployment.

> Please ensure you have a working 

Reviewer 2: This sentence seems incomplete

![image](https://user-images.githubusercontent.com/3837114/95729519-4d2a5f80-0c7d-11eb-88ff-04c442ee4de1.png)

Author: I'm growing old and blind! Will work on the whole thing this weekend! :runner:

.. _getting-to-zero:

Setup Kubernetes
----------------

> This section describes a Kubernetes cluster and outlines how to complete 
> *Step Zero: your Kubernetes cluster* for different cloud providers and infrastructure.
> Since this project utilizes Helm as the package manager for Kubernetes deployments
> defined as charts, we will also cover how to set it up for your infrastructure.

Reviewer 2: ```suggestion
This section describes a how to setup a Kubernetes cluster on a selection of cloud providers and environments, as well as initialize Helm, a Kubernetes package manager, to work with it.
```
